src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -218,5 +221,41 @@ bool BulletPhysicsManager::contactTest(const int physObjectID) {
       ->contactTest();
 }
 
+RaycastResults BulletPhysicsManager::castRay(const esp::geo::Ray& ray,
+                                             double maxDistance) {
+  RaycastResults results;
+  results.ray = ray;
+  double rayLength = ray.direction.length();
+  if (rayLength == 0) {
+    LOG(ERROR) << "BulletPhysicsManager::castRay : Cannot case ray with zero "
+                  "length, aborting. ";
+    return results;
+  }
+  btVector3 from(ray.origin);
+  btVector3 to(ray.origin + ray.direction * maxDistance);
+
+  btCollisionWorld::AllHitsRayResultCallback allResults(from, to);
```

Comment:
Nit: Consider ClosestRayResultCallback here as a second option or instead of AllHits. It is probably good enough for most uses and is presumably faster.

Reply:
Yeah, I guess it comes down to providing access to one vs. all rays. I'm not sure that speed is better without a broad-phase policy to match. We'll consider testing on this and possibly adding a flag later. 👍